### ISSUE TITLE
nonce is stored as paymentId for transaction 8f15be1f355c0b78c79d1cd500cd7e10a0efeb50003e1e74ddd9c62daabea01f

### ISSUE THREAD

**Issue by ExtraHash** [turtlecoin/turtlecoin-utils#40]
It looks like this is actually the nonce of the transaction according to the official explorer, but it's getting parsed as paymentId from utils:

https://explorer.turtle.army/transactions/8f15be1f355c0b78c79d1cd500cd7e10a0efeb50003e1e74ddd9c62daabea01f

Reproduction code:

```ts
import { Transaction } from 'turtlecoin-utils';

const rawTx = '01cb97990101ffa3979901060802f77a50b0b8a62fab2577dbb3fc430a628a4664e7f85f99adb0ff48921146f120d80402db0d581b9b87b17545eebcaf64eb4e38ce6ae33eb3e61b32ac14e81232d8362c882702f481f4648319e933d643acb1a083edc32a2c7b4ac5f52c2d4e98d8c3983ee9d1e0d40302de8373072743f69b251013671c93c4126509deb14251e54eca1ddc737c4c143de0dc2a023ad6941276ad19a293221ed7c32dbc0c9665637573646cc255a4f1764377474880897a02ed9861f2b5055d1cc725853e049a55c4d16c23c341dc9973d7aa2ca6e11e54f65601d628688e2ac557e00f41bbf9b758f0d494368a4ea58bc68f94e5ea028b4a70050233000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000';
const tx = Transaction.from(rawTx);
console.log(tx.paymentId); // 0000000000000000000000000000000000000000000000000000000000000000


```

**Comment by brandonlehmann** [comment]
```text
01 (version)
cb979901 (unlocked)
01 (# of inputs)
ff (input type
  a3979901 (input blockindex)
06 (# of outputs
  08 (output amount)
    02 (output type)
    f77a50b0b8a62fab2577dbb3fc430a628a4664e7f85f99adb0ff48921146f120 (output key)
  d804
    02
    db0d581b9b87b17545eebcaf64eb4e38ce6ae33eb3e61b32ac14e81232d8362c
  8827
    02
    f481f4648319e933d643acb1a083edc32a2c7b4ac5f52c2d4e98d8c3983ee9d1
  e0d403
    02
    de8373072743f69b251013671c93c4126509deb14251e54eca1ddc737c4c143
  de0dc2a
    02
    3ad6941276ad19a293221ed7c32dbc0c9665637573646cc255a4f17643774748
  80897a
    02
    ed9861f2b5055d1cc725853e049a55c4d16c23c341dc9973d7aa2ca6e11e54f6
56 (extra size, 86 bytes)
  01 (public key)
    d628688e2ac557e00f41bbf9b758f0d494368a4ea58bc68f94e5ea028b4a7005 (key)
  02 (nonce)
    33 (nonce size, 51 bytes)
      00 (payment ID subtag == 0x00)
        0000000000000000000000000000000000000000000000000000000000000000 (payment id)
      000000000000000000000000000000000000 (unstructured nonce data)
```

Here in lies the problem with loosely structured data in the TX_EXTRA nonce sub field. 

The very first byte of the nonce field contains the tag for a payment ID (0x00) which signals that the next 32-bytes are a payment ID if enough bytes are remaining to fulfill a 32-byte read (there is). As there is enough bytes to perform the read op it is therefore handled as a payment ID. 

Whether or not core should also handle this as a payment ID is a point of discussion; however, given the format of the field, and the use of sub tags, this is less a problem with utils than a problem with core allowing reservation sizes >= 32 bytes (which is far too many for a pool nonce IMO).

There are three options:

* Ignore payment IDs with all `0`s; which isn't safe as technically that's a valid payment ID
* Ignore all payment IDs with all `0`s only on miner transactions; which may work -- as the vast majority of pools do not load anything other than pool nonce values in the TX_EXTRA nonce field; but again, it's up to them to decide and this project cannot account for every whim.
* Leave it alone here as as that data is properly interpreted and it is up to the whim of the downstream author(s) on whether they accept `0` based payment IDs in their application.

TL;DR: this is not a bug, it is working exactly as intended

**Comment by ExtraHash** [comment]
Why then on your explorer is it correctly labeled as a nonce and not a payment id?

**Comment by brandonlehmann** [comment]
> Why then on your explorer is it correctly labeled as a nonce and not a payment id?

Because the data collection agent over at https://github.com/TurtlePay/blockchain-data-collection-agent does not scan the **blobs**, it uses the metadata information returned from the daemon/core -- which is, as i stated above, either (1) ignoring payment IDs with all 0s (possible and likely) or (2) not reading the tag in the nonce field correctly (unlikely as payment IDs work overall in core). That doesn't mean that core is doing it correctly or as intended.
